src/init.ts

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env -S deno run --no-check --allow-read --allow-write='.'
+const gitignore = `\
+/lib/
+/node_modules/
+`;
+
+const shimFile = `\
+// See https://github.com/fromdeno/deno2node#shimming
+`;
+
+async function createShimFile() {
+  await Deno.mkdir("src/");
+  await Deno.writeTextFile("src/shim.node.ts", shimFile);
+}
+
+async function download(url: URL, target: string) {
+  const response = await fetch(url);
+  await Deno.writeTextFile(target, await response.text());
+}
+
+export async function getVersion(): Promise<string> {
+  const packageUrl = new URL("../package.json", import.meta.url);
+  const response = await fetch(packageUrl);
+  const { version } = await response.json();
+  return version;
+}
+
+async function createPackageJson() {
+  const devDependencies = { deno2node: `~${await getVersion()}` };
+  const scripts = { prepare: "deno2node tsconfig.json" };
+  const packageJson = { scripts, devDependencies };
+  const string = JSON.stringify(packageJson, null, 2);
+  // TODO update existing package.json instead of overwriting it
+  // or just refuse to work in non-empty directories
+  await Deno.writeTextFile("package.json", string);
+}
+
+export async function initializeProject() {
+  const tsconfigUrl = new URL("../tsconfig.json", import.meta.url);
+  await Promise.all([
+    createShimFile(),
+    createPackageJson(),
+    download(tsconfigUrl, "tsconfig.json"),
+    Deno.writeTextFile(".gitignore", gitignore),
+  ]);
```

Comment:
Disk IO is not exactly faster when done in parallel. (Not saying performance things would be measurable here.) Performing stuff sequentially leads to better errors.
```suggestion
  await createShimFile()
  await createPackageJson()
  await download(tsconfigUrl, "tsconfig.json")
  await Deno.writeTextFile(".gitignore", gitignore)
```
Note that it's actually quite common for CLIs to use the sync versions.

Reply:
I think I did not consider that you're doing quite a lot of network IO when I wrote this.